service/evaluation/evaluation.go

```diff
@@ -0,0 +1,853 @@
+	// // Default to an in-memory storage, if nothing was explicitly set
+	// if s.storage == nil {
+	// 	s.storage, err = inmemory.NewStorage()
+	// 	if err != nil {
+	// 		log.Errorf("Could not initialize the storage: %v", err)
+	// 	}
+	// }
+
+	return &s
+}
+
+// SetAuthorizer implements UsesAuthorizer
+func (s *Service) SetAuthorizer(auth api.Authorizer) {
+	s.authorizer = auth
+}
+
+// Authorizer implements UsesAuthorizer
+func (s *Service) Authorizer() api.Authorizer {
+	return s.authorizer
+}
+
+// StartEvaluation is a method implementation of the evaluation interface: It periodically starts the evaluation of a cloud service and the given controls_in_scope (e.g., EUCS OPS-13, EUCS OPS-13.2) in the target_of_evaluation. If no inteval time is given, the default value of 5 minutes is used.
+func (s *Service) StartEvaluation(_ context.Context, req *evaluation.StartEvaluationRequest) (resp *evaluation.StartEvaluationResponse, err error) {
+	var (
+		schedulerTag string
+		interval     int
+		toe          *orchestrator.TargetOfEvaluation
+	)
+
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+	}
+
+	// Set the interval to the default value if not set. If the interval is set to 0, the default interval is used.
+	if req.GetInterval() == 0 {
+		interval = defaultInterval
+	} else {
+		interval = int(req.GetInterval())
+	}
+
+	// Get orchestrator client. The orchestrator client is used to retrieve necessary information from the Orchestrator, such as assessment_results, controls or targets_of_evaluation.
+	err = s.initOrchestratorClient()
+	if err != nil {
+		err = fmt.Errorf("could not set orchestrator client: %v", err)
+		log.Error(err)
+		return nil, status.Errorf(codes.Internal, "%s", err)
+	}
+
+	// Get Target of Evaluation. The Target of Evaluation is retrieved to get the controls_in_scope
+	toe, err = s.orchestratorClient.GetTargetOfEvaluation(context.Background(), &orchestrator.GetTargetOfEvaluationRequest{
+		CloudServiceId: req.GetCloudServiceId(),
+		CatalogId:      req.GetCatalogId(),
+	})
+	if err != nil {
+		err = fmt.Errorf("could not get target of evaluation: %v", err)
+		log.Error(err)
+		return nil, status.Errorf(codes.Internal, "%s", err)
+	}
+
+	log.Info("Starting evaluation ...")
+
+	// Scheduler tags must be unique to find the jobs by tag name
+	s.scheduler.TagsUnique()
+
+	// Add the controls_in_scope of the target_of_evaluation including their sub-controls to the scheduler
+	for _, c := range toe.GetControlsInScope() {
+		var controls []*orchestrator.Control
+
+		// TODO(anatheka): I think we do not need that anymore as we are using the controls from the ToE and these should be ok.
+		// We must get the control from the orchestrator, as we do not know if the request is correct and the parent_ids are probably missing
+		control, err := s.getControl(c.GetCategoryCatalogId(), c.GetCategoryName(), c.GetId())
```

Comment:
As discussed in the call: Only use the sub controls specified in the `controlsInScope` and NOT fetch any (sub)controls from orchestrator.